src/sentry/models/projectownership.py

```diff
@@ -245,12 +255,22 @@ def handle_auto_assignment(cls, project_id, event=None, group=None):
         from sentry.models.user import User
         from sentry.services.hybrid_cloud.user import RpcUser
 
-        # If event is passed in, then this is not called from the force auto-assign API, else it is
-        force_autoassign = True
-        if event:
-            force_autoassign = False
+        enable_force_autoassign = False
+        # Force auto-assign will override an existing manual assignment.
+        if (event is None and logging_extra is None) or force_autoassign:
+            # TODO(Leander): Remove this event/logging_details check. This is currently only missing
+            # from getsentry's admin tools, so we can change that caller to set force_autoassign instead
+            enable_force_autoassign = True
```

Comment:
So to clarify, there is an API in `_admin` that forces auto-assignment even after manual assignment. The previous check was just if `event` was missing.

Now that the task is calling this function as well, I didn't want this to just happen by default, but I also didn't want to query eventstore for the specific `Event` when it was just to enrich some logs. So I added the `logging_extra` param and `force_autoassign`.

I'll add another PR to getsentry to use the `force_autoassign` param and that way we can remove this non-obvious param check.